### DISCUSSION TITLE
Creating a .cur file with hotspot defined from PNG

### DISCUSSION THREAD

**Discussion by meetdilip** [ImageMagick/ImageMagick#6333] [Help]
### ImageMagick version

8:6.9.11.60+dfsg-1.3ubuntu0.22.04.3

### Operating system, version and so on

Ubuntu 22.04

### Description

Hi, I have a PNG file. Trying to convert it into a .cur file to be used in Windows. Would need to find a way to add the hotspot as well. It is a simple mouse pointer image.

I am using Ubuntu 22.04. 

Any help will be great. Thanks.

Edit 1 :

this command gave me a .cur file. But not sure which point is the hotspot. There was no prompt to define one.

`convert point.png point.cur`

**Comment by snibgo** [comment]
This seems to be a limitation in IM for the CUR format.

Looking at `coders/icon.c` functions `WriteICONImage()` and `ReadICONImage()` and https://en.wikipedia.org/wiki/ICO_(file_format), the coder needs fixing for `*.cur" files. The IM code assumes the file is ICO, and doesn't correctly process CUR files. At offset 4, the data is "color planes" for ICO, but "hotspot x-offset" for CUR, etc.
